subsystem/kubeapi_test.go

```diff
@@ -304,68 +328,60 @@ func checkInfraEnvCondition(ctx context.Context, key types.NamespacedName, condi
 
 func getDefaultClusterDeploymentSpec(secretRef *corev1.LocalObjectReference) *hivev1.ClusterDeploymentSpec {
 	return &hivev1.ClusterDeploymentSpec{
-		ClusterName: "test-cluster",
+		ClusterName: clusterDeploymentName,
```

Comment:
IMO the name should be a parameter that comes from the test.
As we saw on other PRs (that were not merged yet), using the same name for all tests will lead to issues when we'll have finalizers in place (finalizers add an additional reconcile loop iteration for clean up).

doesn't have to be a part to this PR.